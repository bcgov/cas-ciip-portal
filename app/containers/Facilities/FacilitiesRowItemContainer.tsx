import React from 'react';
import {Badge} from 'react-bootstrap';
import {graphql, createFragmentContainer, RelayProp} from 'react-relay';
import {
  CiipApplicationStatus,
  FacilitiesRowItemContainer_facilitySearchResult
} from 'FacilitiesRowItemContainer_facilitySearchResult.graphql';
import ApplyButton from '../../components/ApplyButton';
interface Props {
  relay: RelayProp;
  facilitySearchResult: FacilitiesRowItemContainer_facilitySearchResult;
  key: number;
}
const statusBadgeColor: Record<
  CiipApplicationStatus,
  'info' | 'danger' | 'success' | 'warning' | 'primary' | 'secondary'
> = {
  DRAFT: 'warning',
  PENDING: 'info',
  REJECTED: 'danger',
  APPROVED: 'success',
  CHANGES_SUBMITTED: 'primary',
  REQUESTED_CHANGES: 'secondary'
};
export const FacilitiesRowItemComponent: React.FunctionComponent<Props> = props => {
  const {facilitySearchResult} = props;
  const {applicationStatus} = facilitySearchResult;

  return (
    <tr>
      <td>{facilitySearchResult.organisationName}</td>
      <td>{facilitySearchResult.facilityName}</td>
      <td>{facilitySearchResult.facilityMailingAddress}</td>
      <td>{facilitySearchResult.facilityPostalCode}</td>
      <td>{facilitySearchResult.facilityCity}</td>
      <td>
        {' '}
        {applicationStatus ? (
          <>
            <Badge
              pill
              style={{width: '100%'}}
              variant={statusBadgeColor[applicationStatus]}
            >
              {applicationStatus}
            </Badge>
          </>
        ) : (
          <>Application not started</>
        )}
      </td>
      <td>
        <ApplyButton applyButtonDetails={props.facilitySearchResult} />
      </td>
    </tr>
  );
};

export default createFragmentContainer(FacilitiesRowItemComponent, {
  facilitySearchResult: graphql`
    fragment FacilitiesRowItemContainer_facilitySearchResult on FacilitySearchResult {
      ...ApplyButton_applyButtonDetails
      facilityName
      facilityMailingAddress
      facilityCity
      facilityPostalCode
      applicationStatus
      organisationName
    }
  `
});
