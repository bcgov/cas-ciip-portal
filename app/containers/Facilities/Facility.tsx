import React from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {Dropdown} from 'react-bootstrap';

export const FacilityComponent = (props) => {
  const {facility} = props;
  if (!facility) {
    return null;
  }

  if (
    !facility.facilityName
      .toLowerCase()
      .includes(props.facilityInput.toLowerCase())
  )
    return null;

  return (
    <Dropdown.Item
      onSelect={() =>
        props.selectFacility(
          facility.facilityName,
          facility.rowId,
          facility.organisationByOrganisationId.operatorName,
          facility.facilityType,
          facility.bcghgid,
          facility.swrsReportId
        )
      }
    >
      {facility.facilityName}
    </Dropdown.Item>
  );
};

export default createFragmentContainer(FacilityComponent, {
  facility: graphql`
    fragment Facility_facility on Facility {
      rowId
      facilityName
      organisationByOrganisationId {
        operatorName
      }
      facilityType
      bcghgid
      swrsReportId
    }
  `
});
