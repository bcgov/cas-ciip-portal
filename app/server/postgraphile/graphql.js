const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
