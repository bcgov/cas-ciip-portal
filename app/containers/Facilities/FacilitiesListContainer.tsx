import React, {useEffect} from 'react';
import {graphql, createRefetchContainer} from 'react-relay';
import SearchTableLayout from 'components/SearchTableLayout';
import FacilitiesRowItemContainer from './FacilitiesRowItemContainer';

export const FacilitiesList = props => {
  const {
    direction,
    orderByField,
    searchField,
    searchValue,
    handleEvent
  } = props;
  const {edges} = props.query.searchAllFacilities;
  useEffect(() => {
    const refetchVariables = {
      searchField,
      searchValue,
      orderByField,
      direction
    };
    props.relay.refetch(refetchVariables);
  });

  const displayNameToColumnNameMap = {
    'Organisation Name': 'organisation_name',
    'Facility Name': 'facility_name',
    Address: 'facility_mailing_address',
    'Postal Code': 'facility_postal_code',
    City: 'facility_city',
    Status: 'application_status'
  };

  const body = (
    <tbody>
      {edges.map(edge => (
        <FacilitiesRowItemContainer
          key={edge.node.rowId}
          facilitySearchResult={edge.node}
        />
      ))}
    </tbody>
  );

  return (
    <SearchTableLayout
      body={body}
      displayNameToColumnNameMap={displayNameToColumnNameMap}
      handleEvent={handleEvent}
    />
  );
};

export default createRefetchContainer(
  FacilitiesList,
  {
    query: graphql`
      fragment FacilitiesListContainer_query on Query
        @argumentDefinitions(
          searchField: {type: "String"}
          searchValue: {type: "String"}
          orderByField: {type: "String"}
          direction: {type: "String"}
        ) {
        searchAllFacilities(
          searchField: $searchField
          searchValue: $searchValue
          orderByField: $orderByField
          direction: $direction
        ) {
          edges {
            node {
              rowId
              ...FacilitiesRowItemContainer_facilitySearchResult
            }
          }
        }
      }
    `
  },
  graphql`
    query FacilitiesListContainerRefetchQuery(
      $searchField: String
      $searchValue: String
      $orderByField: String
      $direction: String
    ) {
      query {
        ...FacilitiesListContainer_query
          @arguments(
            orderByField: $orderByField
            direction: $direction
            searchField: $searchField
            searchValue: $searchValue
          )
      }
    }
  `
);
