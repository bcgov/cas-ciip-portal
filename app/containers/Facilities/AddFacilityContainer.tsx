import React, {useState} from 'react';
import {graphql, createFragmentContainer, RelayProp} from 'react-relay';
import {Button, Modal} from 'react-bootstrap';
import {JSONSchema6} from 'json-schema';
import JsonSchemaForm, {IChangeEvent} from 'react-jsonschema-form';
import moment from 'moment-timezone';
import {AddFacilityContainer_organisation} from 'AddFacilityContainer_organisation.graphql';
import FormFieldTemplate from '../Forms/FormFieldTemplate';
import FormObjectFieldTemplate from '../Forms/FormObjectFieldTemplate';

interface Props {
  relay: RelayProp;
  organisation: AddFacilityContainer_organisation;
}

export const AddFacility = props => {
  const {organisation} = props;
  const [modalShow, setModalShow] = useState(false);
  const addFacilitySchema: JSONSchema6 = {
    type: 'object',
    title: 'Facility Details',
    properties: {
      facilityName: {
        title: 'Facility Name',
        type: 'string'
      },
      facilityType: {
        type: 'string',
        title: 'Facility Type',
        enum: ['IF_a', 'IF_b', 'L_c', 'SFO', 'LFO', 'EIO']
      },
      bcghgid: {
        type: 'string',
        title: 'BCGHG ID Number'
      },
      naics: {
        type: 'string',
        title: 'NAICS Code'
      },
      mailingAddress: {
        title: 'Facility Mailing Address',
        $ref: '#/definitions/address'
      }
    },
    definitions: {
      address: {
        type: 'object',
        properties: {
          streetAddress: {
            title: 'Street address',
            type: 'string'
          },
          city: {
            title: 'City',
            type: 'string'
          },
          province: {
            title: 'Province or Territory',
            type: 'string',
            enum: [
              'Alberta',
              'British Columbia',
              'Manitoba',
              'New Brunswick',
              'Newfoundland and Labrador',
              'Northwest Territories',
              'Nova Scotia',
              'Nunavut',
              'Ontario',
              'Prince Edward Island',
              'Quebec',
              'Saskatchewan',
              'Yukon'
            ]
          },
          postalCode: {
            title: 'Postal Code',
            type: 'string',
            format: 'postal-code'
          }
        }
      }
    }
  };

  const customFormats = {
    'postal-code': /[A-Z]\d[A-Z]\s?\d[A-Z]\d/
  };

  const addFacilityUISchema = {
    benchmark: {
      'ui:col-md': 6
    },
    description: {
      'ui:col-md': 6
    },
    startReportingYear: {
      'ui:col-md': 6,
      'ui:help': 'The first reporting year where this benchmark is used'
    },
    endReportingYear: {
      'ui:col-md': 6,
      'ui:help': 'The last reporting year where this benchmark is used'
    }
  };

  const currentYear = moment().format('YYYY');

  const saveNewFacility = (e: IChangeEvent) => {
    // Const {environment} = props.relay;
    const variables = {
      ...e.formData,
      currentYear,
      organisationId: organisation.rowId
    };
    console.log(variables);
  };

  return (
    <>
      <Button onClick={() => setModalShow(!modalShow)}>Add Facility +</Button>
      <Modal
        centered
        size="xl"
        show={modalShow}
        onHide={() => setModalShow(false)}
      >
        <Modal.Header>Add Facility</Modal.Header>
        <Modal.Body>
          <JsonSchemaForm
            schema={addFacilitySchema}
            uiSchema={addFacilityUISchema}
            customFormats={customFormats}
            showErrorList={false}
            FieldTemplate={FormFieldTemplate}
            ObjectFieldTemplate={FormObjectFieldTemplate}
            onSubmit={saveNewFacility}
          >
            <Button type="submit" variant="primary">
              Save Facility
            </Button>
          </JsonSchemaForm>
        </Modal.Body>
      </Modal>
    </>
  );
};

export default createFragmentContainer(AddFacility, {
  organisation: graphql`
    fragment AddFacilityContainer_organisation on Organisation {
      rowId
    }
  `
});
