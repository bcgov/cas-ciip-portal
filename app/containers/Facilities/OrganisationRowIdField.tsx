import React, {useMemo} from 'react';
import {FieldProps} from 'react-jsonschema-form';
import {createFragmentContainer, graphql} from 'react-relay';
import {OrganisationRowIdField_query} from 'OrganisationRowIdField_query.graphql';

interface Props extends FieldProps<number> {
  query: OrganisationRowIdField_query;
}

/**
 * Fetches the list of fuels and their row ids
 */
export const OrganisationRowIdFieldComponent: React.FunctionComponent<Props> = (
  props
) => {
  console.log(props);
  const fieldProps = useMemo(
    () => ({
      ...props,
      schema: {
        ...props.schema,
        enum: props.query.allOrganisations.edges.map(({node}) => node.rowId),
        enumNames: props.query.allOrganisations.edges.map(
          ({node}) => node.operatorName
        )
      }
    }),
    [props]
  );

  return <props.registry.fields.StringField {...fieldProps} />;
};

export default createFragmentContainer(OrganisationRowIdFieldComponent, {
  query: graphql`
    fragment OrganisationRowIdField_query on Query {
      allOrganisations {
        edges {
          node {
            rowId
            operatorName
          }
        }
      }
    }
  `
});
