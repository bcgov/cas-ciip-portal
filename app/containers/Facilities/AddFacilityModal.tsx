import React, {useState} from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {AddFacilityModal_query} from 'AddFacilityModal_query.graphql';
import {Button, Modal, Card} from 'react-bootstrap';
import JsonSchemaForm, {IChangeEvent} from '@rjsf/core';
import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
import addFacilitySchema from 'components/facility/addFacilitySchema';
import OrganisationRowIdField from './OrganisationRowIdField';
import SearchDropdownWidget from 'components/Forms/SearchDropdownWidget';
import NumberField from 'containers/Forms/NumberField';

interface Props {
  query: AddFacilityModal_query;
  onAddFacility?: (...args: any[]) => void;
}

const AddFacilityModal: React.FunctionComponent<Props> = (props) => {
  const {onAddFacility, query} = props;
  const [isModalVisible, setModalVisible] = useState(false);

  const customFormats = {
    'postal-code': /[a-z]\d[a-z]\s?\d[a-z]\d/i
  };

  const CUSTOM_FIELDS = {
    organisationRowId: (props) => (
      <OrganisationRowIdField query={props.formContext.query} {...props} />
    ),
    NumberField
  };

  const addFacilityUiSchema = {
    organisationRowId: {
      'ui:col-md': 6,
      'ui:widget': 'SearchWidget',
      'ui:field': 'organisationRowId'
    }
  };

  const saveNewFacility = async (e: IChangeEvent) => {
    const {facilityName, facilityType, bcghgid, organisationRowId} = e.formData;
    const variables = {
      input: {
        facility: {
          facilityName,
          facilityType,
          bcghgid,
          organisationId: organisationRowId
        }
      }
    };
    setModalVisible(!isModalVisible);
    console.log(variables);
    onAddFacility(variables);
  };

  return (
    <>
      <Card style={{marginTop: '50px'}}>
        <Card.Body>
          <Card.Title>
            Can&apos;t find the Facility you&apos;re looking for?
          </Card.Title>
          <Card.Text>
            You can add a facility if it is new or hasn&apos;t reported before.
          </Card.Text>
          <Button
            variant="outline-primary"
            onClick={() => setModalVisible(!isModalVisible)}
          >
            Add a new Facility
          </Button>
        </Card.Body>
      </Card>
      <Modal
        centered
        size="lg"
        show={isModalVisible}
        onHide={() => setModalVisible(false)}
      >
        <Modal.Header>
          <h5>Add a new Facility</h5>
        </Modal.Header>
        <Modal.Body>
          <JsonSchemaForm
            schema={addFacilitySchema}
            uiSchema={addFacilityUiSchema}
            customFormats={customFormats}
            formContext={{query}}
            fields={CUSTOM_FIELDS}
            widgets={{SearchWidget: SearchDropdownWidget}}
            showErrorList={false}
            FieldTemplate={FormFieldTemplate}
            ObjectFieldTemplate={FormObjectFieldTemplate}
            onSubmit={saveNewFacility}
          >
            <div>
              <Button
                style={{marginRight: '10px'}}
                type="submit"
                variant="primary"
              >
                Save
              </Button>
              <Button
                variant="danger"
                onClick={() => setModalVisible(!isModalVisible)}
              >
                Close
              </Button>
            </div>
          </JsonSchemaForm>
        </Modal.Body>
      </Modal>
    </>
  );
};

export default createFragmentContainer(AddFacilityModal, {
  query: graphql`
    fragment AddFacilityModal_query on Query {
      ...OrganisationRowIdField_query
    }
  `
});
