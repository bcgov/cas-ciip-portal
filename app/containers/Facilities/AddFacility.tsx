import React, { useState } from "react";
import { graphql, createFragmentContainer, RelayProp } from "react-relay";
import { Dropdown, FormControl, Card, Table } from "react-bootstrap";
import { AddFacility_query } from "AddFacility_query.graphql";
import AddFacilityModal from "./AddFacilityModal";
import createFacilityMutation from "mutations/facility/createFacilityMutation";
import Facility from "./Facility";

interface Props {
  query: AddFacility_query;
  relay: RelayProp;
  facilityInput: string;
  selectedFacility: number;
  handleInputChange: (event: any) => void;
  handleFacilityChange: (event: any) => void;
}

export const AddFacilityComponent: React.FunctionComponent<Props> = (props) => {
  const [selectedFacilityDetails, setSelectedFacilityDetails] = useState(null);
  const { query } = props;
  const { allFacilities } = query;
  const changeInput = (event) => {
    event.stopPropagation();
    event.preventDefault();
    event.persist();
    props.handleInputChange(event.target.value);
  };

  const selectFacility = (
    name,
    id,
    operatorName,
    facilityType,
    bcghgid,
    swrsReportId
  ) => {
    props.handleFacilityChange(id);
    props.handleInputChange("");
    setSelectedFacilityDetails({
      name,
      id,
      operatorName,
      facilityType,
      bcghgid,
      swrsReportId,
    });
  };

  const handleAddFacility = async (variables) => {
    const { environment } = props.relay;
    await createFacilityMutation(environment, variables);
  };

  return (
    <div>
      <Card style={{ marginTop: "50px" }}>
        <Card.Body>
          <Card.Title>Search for a facility: </Card.Title>
          <Dropdown className="search-dropdown">
            <Dropdown.Toggle id="org-dropdown" className="search-toggle">
              Find Facility
            </Dropdown.Toggle>
            <Dropdown.Menu>
              <FormControl
                placeholder="Search..."
                className="mx-3 my-2 w-auto"
                onChange={changeInput}
              />
              <div className="facility-scroll">
                {allFacilities.edges.map(({ node }) => {
                  return (
                    <Facility
                      key={node.id}
                      select
                      facility={node}
                      facilityInput={props.facilityInput}
                      selectFacility={selectFacility}
                    />
                  );
                })}
              </div>
            </Dropdown.Menu>
          </Dropdown>
          {props.selectedFacility !== null && (
            <Table
              style={{
                textAlign: "center",
                marginTop: "10px",
                borderBottom: "1px solid rgb(0,0,0,0.1)",
              }}
            >
              <thead>
                <tr>
                  <th>Operator Name</th>
                  <th>Facility Name</th>
                  <th>Facility Type</th>
                  <th>BCGHGID</th>
                  <th>SWRS Report ID</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>{selectedFacilityDetails?.operatorName}</td>
                  <td>{selectedFacilityDetails?.name}</td>
                  <td>{selectedFacilityDetails?.facilityType}</td>
                  <td>{selectedFacilityDetails?.bcghgid}</td>
                  <td>{selectedFacilityDetails?.swrsReportId}</td>
                </tr>
              </tbody>
            </Table>
          )}
          <AddFacilityModal query={query} onAddFacility={handleAddFacility} />
          <style jsx>
            {`
              .facility-scroll {
                max-height: 250px;
                overflow: hidden;
                overflow-y: scroll;
              }

              thead {
                background: #036;
                color: #fff;
              }
            `}
          </style>
        </Card.Body>
      </Card>
    </div>
  );
};

export default createFragmentContainer(AddFacilityComponent, {
  query: graphql`
    fragment AddFacility_query on Query {
      allFacilities {
        edges {
          node {
            id
            rowId
            ...Facility_facility
          }
        }
      }
      ...AddFacilityModal_query
    }
  `,
});
