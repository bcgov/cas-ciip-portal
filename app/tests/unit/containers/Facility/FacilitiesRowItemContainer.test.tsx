import React from 'react';
import {shallow} from 'enzyme';
import {FacilitiesRowItemContainer_facilitySearchResult} from 'FacilitiesRowItemContainer_facilitySearchResult.graphql';
import {FacilitiesRowItemContainer_query} from 'FacilitiesRowItemContainer_query.graphql';
import {FacilitiesRowItemComponent} from 'containers/Facilities/FacilitiesRowItemContainer';

describe('FacilitiesRowItem', () => {
  it('should match the previous snapshot', async () => {
    const facilitySearchResult: FacilitiesRowItemContainer_facilitySearchResult = {
      ' $fragmentRefs': {
        ApplyButtonContainer_applyButtonDetails: true
      },
      ' $refType': 'FacilitiesRowItemContainer_facilitySearchResult',
      facilityName: 'fac',
      facilityType: 'EIO',
      facilityBcghgid: '123',
      lastSwrsReportingYear: 2019,
      applicationRevisionStatus: 'SUBMITTED',
      organisationName: 'org',
      applicationByApplicationId: {
        latestDraftRevision: {
          certificationUrl: {
            __typename: 'certificationUrl'
          }
        }
      }
    };
    const query: FacilitiesRowItemContainer_query = {
      ' $fragmentRefs': {
        ApplyButtonContainer_query: true
      },
      ' $refType': 'FacilitiesRowItemContainer_query'
    };

    const r = shallow(
      <FacilitiesRowItemComponent
        facilitySearchResult={facilitySearchResult}
        query={query}
        reportingYear={2019}
      />
    );
    expect(r).toMatchSnapshot();
  });

  it('should show Pending Certification in the status if the application is in draft and a certificationUrl exists', async () => {
    const facilitySearchResult: FacilitiesRowItemContainer_facilitySearchResult = {
      ' $fragmentRefs': {
        ApplyButtonContainer_applyButtonDetails: true
      },
      ' $refType': 'FacilitiesRowItemContainer_facilitySearchResult',
      facilityName: 'fac',
      facilityType: 'EIO',
      facilityBcghgid: '123',
      lastSwrsReportingYear: 2019,
      applicationRevisionStatus: 'DRAFT',
      organisationName: 'org',
      applicationByApplicationId: {
        latestDraftRevision: {
          certificationUrl: {
            __typename: 'certificationUrl'
          }
        }
      }
    };
    const query: FacilitiesRowItemContainer_query = {
      ' $fragmentRefs': {
        ApplyButtonContainer_query: true
      },
      ' $refType': 'FacilitiesRowItemContainer_query'
    };

    const r = shallow(
      <FacilitiesRowItemComponent
        facilitySearchResult={facilitySearchResult}
        query={query}
        reportingYear={2019}
      />
    );
    expect(r).toMatchSnapshot();
    expect(r.find('Badge').text()).toBe('DRAFT (Pending Certification)');
  });

  it('should not show Pending Certification in the status if the application is in draft but a certificationUrl does not exist', async () => {
    const facilitySearchResult: FacilitiesRowItemContainer_facilitySearchResult = {
      ' $fragmentRefs': {
        ApplyButtonContainer_applyButtonDetails: true
      },
      ' $refType': 'FacilitiesRowItemContainer_facilitySearchResult',
      facilityName: 'fac',
      facilityType: 'EIO',
      facilityBcghgid: '123',
      lastSwrsReportingYear: 2019,
      applicationRevisionStatus: 'DRAFT',
      organisationName: 'org',
      applicationByApplicationId: {
        latestDraftRevision: {
          certificationUrl: null
        }
      }
    };
    const query: FacilitiesRowItemContainer_query = {
      ' $fragmentRefs': {
        ApplyButtonContainer_query: true
      },
      ' $refType': 'FacilitiesRowItemContainer_query'
    };

    const r = shallow(
      <FacilitiesRowItemComponent
        facilitySearchResult={facilitySearchResult}
        query={query}
        reportingYear={2019}
      />
    );
    expect(r).toMatchSnapshot();
    expect(r.find('Badge').text()).toBe('DRAFT');
  });

  it('should not show Pending Certification in the status if the application is not in draft state and a certificationUrl exists', async () => {
    const facilitySearchResult: FacilitiesRowItemContainer_facilitySearchResult = {
      ' $fragmentRefs': {
        ApplyButtonContainer_applyButtonDetails: true
      },
      ' $refType': 'FacilitiesRowItemContainer_facilitySearchResult',
      facilityName: 'fac',
      facilityType: 'EIO',
      facilityBcghgid: '123',
      lastSwrsReportingYear: 2019,
      applicationRevisionStatus: 'APPROVED',
      organisationName: 'org',
      applicationByApplicationId: {
        latestDraftRevision: {
          certificationUrl: {
            __typename: 'certificationUrl'
          }
        }
      }
    };
    const query: FacilitiesRowItemContainer_query = {
      ' $fragmentRefs': {
        ApplyButtonContainer_query: true
      },
      ' $refType': 'FacilitiesRowItemContainer_query'
    };

    const r = shallow(
      <FacilitiesRowItemComponent
        facilitySearchResult={facilitySearchResult}
        query={query}
        reportingYear={2019}
      />
    );
    expect(r).toMatchSnapshot();
    expect(r.find('Badge').text()).toBe('APPROVED');
  });
});
