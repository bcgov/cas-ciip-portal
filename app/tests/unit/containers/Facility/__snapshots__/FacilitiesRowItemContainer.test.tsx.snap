// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilitiesRowItem should match the previous snapshot 1`] = `
<tr>
  <td>
    org
  </td>
  <td>
    fac
  </td>
  <td>
    EIO
  </td>
  <td>
    123
  </td>
  <td>
    2019
  </td>
  <td>
     
    <Badge
      pill={true}
      style={
        Object {
          "width": "100%",
        }
      }
      variant="info"
    >
      SUBMITTED
    </Badge>
  </td>
  <td>
    <Relay(ApplyButton)
      applyButtonDetails={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_applyButtonDetails": true,
          },
          " $refType": "FacilitiesRowItemContainer_facilitySearchResult",
          "applicationByApplicationId": Object {
            "latestDraftRevision": Object {
              "certificationUrl": Object {
                "id": "abc",
              },
            },
          },
          "applicationRevisionStatus": "SUBMITTED",
          "facilityBcghgid": "123",
          "facilityName": "fac",
          "facilityType": "EIO",
          "lastSwrsReportingYear": 2019,
          "organisationName": "org",
        }
      }
      query={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_query": true,
          },
          " $refType": "FacilitiesRowItemContainer_query",
        }
      }
    />
  </td>
</tr>
`;

exports[`FacilitiesRowItem should not show Pending Certification in the status if the application is in draft but a certificationUrl does not exist 1`] = `
<tr>
  <td>
    org
  </td>
  <td>
    fac
  </td>
  <td>
    EIO
  </td>
  <td>
    123
  </td>
  <td>
    2019
  </td>
  <td>
     
    <Badge
      pill={true}
      style={
        Object {
          "width": "100%",
        }
      }
      variant="warning"
    >
      DRAFT
    </Badge>
  </td>
  <td>
    <Relay(ApplyButton)
      applyButtonDetails={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_applyButtonDetails": true,
          },
          " $refType": "FacilitiesRowItemContainer_facilitySearchResult",
          "applicationByApplicationId": Object {
            "latestDraftRevision": Object {
              "certificationUrl": null,
            },
          },
          "applicationRevisionStatus": "DRAFT",
          "facilityBcghgid": "123",
          "facilityName": "fac",
          "facilityType": "EIO",
          "lastSwrsReportingYear": 2019,
          "organisationName": "org",
        }
      }
      query={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_query": true,
          },
          " $refType": "FacilitiesRowItemContainer_query",
        }
      }
    />
  </td>
</tr>
`;

exports[`FacilitiesRowItem should not show Pending Certification in the status if the application is not in draft state and a certificationUrl exists 1`] = `
<tr>
  <td>
    org
  </td>
  <td>
    fac
  </td>
  <td>
    EIO
  </td>
  <td>
    123
  </td>
  <td>
    2019
  </td>
  <td>
     
    <Badge
      pill={true}
      style={
        Object {
          "width": "100%",
        }
      }
      variant="success"
    >
      APPROVED
    </Badge>
  </td>
  <td>
    <Relay(ApplyButton)
      applyButtonDetails={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_applyButtonDetails": true,
          },
          " $refType": "FacilitiesRowItemContainer_facilitySearchResult",
          "applicationByApplicationId": Object {
            "latestDraftRevision": Object {
              "certificationUrl": Object {
                "id": "abc",
              },
            },
          },
          "applicationRevisionStatus": "APPROVED",
          "facilityBcghgid": "123",
          "facilityName": "fac",
          "facilityType": "EIO",
          "lastSwrsReportingYear": 2019,
          "organisationName": "org",
        }
      }
      query={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_query": true,
          },
          " $refType": "FacilitiesRowItemContainer_query",
        }
      }
    />
  </td>
</tr>
`;

exports[`FacilitiesRowItem should show Pending Certification in the status if the application is in draft and a certificationUrl exists 1`] = `
<tr>
  <td>
    org
  </td>
  <td>
    fac
  </td>
  <td>
    EIO
  </td>
  <td>
    123
  </td>
  <td>
    2019
  </td>
  <td>
     
    <Badge
      pill={true}
      style={
        Object {
          "width": "100%",
        }
      }
      variant="warning"
    >
      DRAFT
       (Pending Certification)
    </Badge>
  </td>
  <td>
    <Relay(ApplyButton)
      applyButtonDetails={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_applyButtonDetails": true,
          },
          " $refType": "FacilitiesRowItemContainer_facilitySearchResult",
          "applicationByApplicationId": Object {
            "latestDraftRevision": Object {
              "certificationUrl": Object {
                "id": "abc",
              },
            },
          },
          "applicationRevisionStatus": "DRAFT",
          "facilityBcghgid": "123",
          "facilityName": "fac",
          "facilityType": "EIO",
          "lastSwrsReportingYear": 2019,
          "organisationName": "org",
        }
      }
      query={
        Object {
          " $fragmentRefs": Object {
            "ApplyButtonContainer_query": true,
          },
          " $refType": "FacilitiesRowItemContainer_query",
        }
      }
    />
  </td>
</tr>
`;
